File: UI/grooveon_desktop/lib/models/response/answer_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'answer_response.g.dart';

@JsonSerializable()
class AnswerResponse {
  final int id;
  final int questionId;
  final String? questionTitle;
  final int adminId;
  final String? adminUserName;
  final String message;
  final DateTime createdAt;

  AnswerResponse({
    required this.id,
    required this.questionId,
    this.questionTitle,
    required this.adminId,
    this.adminUserName,
    required this.message,
    required this.createdAt,
  });

  factory AnswerResponse.fromJson(Map<String, dynamic> json) =>
      _$AnswerResponseFromJson(json);

  Map<String, dynamic> toJson() => _$AnswerResponseToJson(this);
}

File: GrooveOn.Models/ResponseObjects/AnswerResponse.cs
using System;

namespace GrooveOn.Model.ResponseObjects
{
    public class AnswerResponse
    {
        public int Id { get; set; }

        public int QuestionId { get; set; }
        public string? QuestionTitle { get; set; }

        public int AdminId { get; set; }
        public string? AdminUserName { get; set; }

        public string Message { get; set; } = string.Empty;

        public DateTime CreatedAt { get; set; }
    }
}